manual/source/utilities.rst

```diff
@@ -328,10 +347,128 @@ low level programmatic access to the data structures.
 **HDFview** (http://www.hdfgroup.org)
     A Java based GUI for browsing (and some basic plotting) of HDF files.
 
-Language APIs
--------------
+.. _language.apis:
+
+Language APIs for NeXus and HDF5
+################################
+
+Collected here are some of the tools identified [#]_ as a result
+of a simple question asked at the 2018 Nobugs conference:
+*Are there examples of code that reads NeXus data?*
+Some of these are very specific to an instrument or application
+definition while others are more generic.
+The lists below are organized by programming language,
+yet some collections span several languages so they are
+listed in the section :ref:`language.apis.mixed`.
+
+.. [#] https://github.com/nexusformat/definitions/issues/630
+
+
+.. _language.apis.f77:
+
+Language API: *F77*
++++++++++++++++++++
+
+* **POLDI**: ``poldi.zip`` [#]_ contains:
+  - A F77 reading routine using NAPI for POLDI at SINQ PSI
+  - an example of a file which it reads
+
+  .. [#] https://github.com/nexusformat/definitions/files/4107360/poldi.zip
+
+
+.. _language.apis.idl:
+
+Language API: *IDL*
++++++++++++++++++++
+
+* **aXis2000** [#]_, with the NeXus-specific IDL code 
+  in the ``read_nexus.pro`` [#]_, reads :ref:`NXstxm`
+
+  .. [#] http://unicorn.chemistry.mcmaster.ca/aXis2000.html
+  .. [#] ``read_nexus.pro``: http://unicorn.chemistry.mcmaster.ca/axis/aXis2000.zip
+
+
+.. _language.apis.igorpro:
+
+Language API: *IgorPro*
++++++++++++++++++++++++
+
+* **HDF5gateway** [#]_ makes it easy to read a HDF5 file (including NeXus) into an 
+  IgorPro [#]_ folder, including group and dataset attributes, 
+  such as a NeXus data file, modify it, and then write 
+  the folder structure back out.
+
+  .. [#] https://github.com/prjemian/hdf5gateway
+  .. [#] IgorPro: https://wavemetrics.com
+
+.. _language.apis.java:
 
-**h5py** (http://docs.h5py.org/)
+Language API: *Java*
+++++++++++++++++++++
+
+* **Dawn** [#]_ has java code to read [#]_ and write [#]_ 
+  HDF5 NeXus files (generic NeXus, not tied to 
+  specific application definitions). 
+
+  .. [#] https://dawnsci.org/
+  .. [#] read: https://github.com/DawnScience/scisoft-core/blob/master/uk.ac.diamond.scisoft.analysis/src/uk/ac/diamond/scisoft/analysis/io/NexusHDF5Loader.java
+  .. [#] write: https://github.com/DawnScience/dawnsci/blob/master/org.eclipse.dawnsci.hdf5/src/org/eclipse/dawnsci/hdf5/nexus/NexusFileHDF5.java
+
+* ``NXreader.zip`` [#]_ is java code which reads NeXus files into **ImageJ.** 
+  It uses the Java-hdf interface to HDF5. It tries to do a good 
+  job locating the image dataset by NeXus conventions. 
+  But it uses the old style conventions. 
+
+  .. [#] https://github.com/nexusformat/definitions/files/4107439/NXreader.zip
+
+
+.. _language.apis.python:
+
+Language API: *Python*
+++++++++++++++++++++++
+
+* **Dials** [#]_ has code for reading :ref:`NXmx` [#]_
```

Comment:
Maybe like this:

* **Dials** [#]_ has python (and some C++) code for reading :ref:`NXmx` [#]_